projects/core/src/auth/user-auth/services/auth-http-header.service.ts

```diff
@@ -59,7 +59,7 @@ export class AuthHttpHeaderService {
     return rawValue;
   }
 
-  protected createAuthorizationHeader(): { Authorization?: string } {
+  protected createAuthorizationHeader(): { Authorization: string } | {} {
```

Comment:
Comment for reviewer: could the type tweaks made in all files from `/project/core` be considered breaking changes?

Reply:
While `{ Authorization?: string }` and `{ Authorization: string } | {}` seem equivalent, `{ Authorization?: string }` does not seem to comply, in strict mode,  with type `{ [name: string]: string | string[]; }` where the return value is assigned to.

It also seems that there is a bit more to handling this case where the function is called than simply casting the result.

I presume we could store the function result in a variable and then check if it has some keys before deciding which value is passed along in the http headers.  It feels like we would be decentralizing the logic that goes into generating the authorization header.

Reply:
We agree to stick with `{ Authorization: string } | {}`